src/os_win32.c

```diff
@@ -3640,6 +3708,8 @@ mch_exit(int r)
     else
 	mch_exit_c(r);
 #elif defined(FEAT_GUI_MSWIN)
+    if (vimrun_path_allocated)
+	vim_free(vimrun_path.string);
```

Comment:
Originally, `vimrun_path` was allocated in `mch_init_g()` only once and its lifetime was the same as the gvim.exe process. So, it was not freed explicitly. So, this part may not be needed.

But, if you want to free it explicitly, it is better to add the same code in the `#ifdef VIMDLL` part above.

Reply:
When `VIMDLL` is defined, `FEAT_GUI_MSWIN` is also defined.

`VIMDLL` creates a DLL that supports both GUI and console.